docs/tutorials/rust.md

```diff
@@ -0,0 +1,152 @@
+```shell
+pixi add rust
+```
+
+This will add the `rust` package to your `pixi.toml` file.
+
+## Add a `cargo` project
+As you now have rust installed, you can create a `cargo` project in your `pixi` project.
+```shell
+pixi run cargo init
+```
+
+Now we can build a `cargo` project using `pixi`.
+```shell
+pixi run cargo build
+```
+To simplify the build process, you can add a `build` task to your `pixi.toml` file using the following command:
+```shell
+pixi task add build "cargo build"
+```
+Which creates this field in the `pixi.toml` file:
+```toml title="pixi.toml"
+[tasks]
+build = "cargo build"
+```
+
+And now you can build your project using:
+```shell
+pixi run build
+```
+
+You can also run your project using:
+```shell
+pixi run cargo run
+```
+Which you can simplify with a task again.
+```shell
+pixi task add start "cargo run"
+```
+
+So you should get the following output:
+```shell
+pixi run start
+Hello, world!
+```
+
+Congratulations you have a Rust project running on your machine with pixi!
+
+## Next steps, why is this useful when there is `rustup`?
+Cargo is not a binary package manager, but a source package manager.
+This means that you need to have the Rust compiler installed on your system to use it.
+And possibly other dependencies that are not included in the `cargo` package manager.
+For example, you might need to install `openssl` or `libssl-dev` on your system to build a package.
+This is the case for `pixi` as well, but `pixi` will install these dependencies in your project folder, so you don't have to worry about them.
+
+Add the following dependencies to your cargo project:
+```shell
+pixi run cargo add git2
+```
+
+If your system is not preconfigured to build C and have the `libssl-dev` package installed you will not be able to build the project:
+```shell
+pixi run build
+...
+Could not find directory of OpenSSL installation, and this `-sys` crate cannot
+proceed without this knowledge. If OpenSSL is installed and this crate had
+trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the
+compilation process.
+
+Make sure you also have the development packages of openssl installed.
+For example, `libssl-dev` on Ubuntu or `openssl-devel` on Fedora.
+
+If you're in a situation where you think the directory *should* be found
+automatically, please open a bug at https://github.com/sfackler/rust-openssl
+and include information about your system as well as this message.
+
+$HOST = x86_64-unknown-linux-gnu
+$TARGET = x86_64-unknown-linux-gnu
+openssl-sys = 0.9.102
+
+
+It looks like you're compiling on Linux and also targeting Linux. Currently this
+requires the `pkg-config` utility to find OpenSSL but unfortunately `pkg-config`
+could not be found. If you have OpenSSL installed you can likely fix this by
+installing `pkg-config`.
+...
+```
+You can fix this with pixi:
```

Comment:
maybe we can write something like this:
You can fix this by adding necessary dependencies for building git2